### PR TITLE
Transformation: Add variable support for filter by value regex matcher

### PR DESCRIPTION
**Pull request by nmarrs**
Add support for template variable interpolation for the regex matcher.

Before

https://github.com/user-attachments/assets/27331226-fa5d-4b34-b2b3-d5096187cf44

After 

https://github.com/user-attachments/assets/967ea5ff-b07a-4d29-b52d-4d7731a7249c

[Filter by value transformation regex variable support-1721896735727.json](https://github.com/user-attachments/files/16373088/Filter.by.value.transformation.regex.variable.support-1721896735727.json)

Fixes https://github.com/grafana/support-escalations/issues/11615

### CODE REVIEW

**Comment by samjewell** [comment]
Looks great, thanks for this change.
Sorry to ask - does this support Multi-select / select-All?

File: packages/grafana-data/src/transformations/transformers/filterByValue.ts [hunk 1]
```diff
@@ -78,8 +78,16 @@ export const filterByValueTransformer: DataTransformerInfo<FilterByValueTransfor
               },
             };
           } else if (filter.config.id === ValueMatcherID.regex) {
-            // Due to colliding syntaxes, interpolating regex filters will cause issues.
-            return filter;
+            // Due to colliding syntaxes, interpolating regex filters may cause issues.
+            let value = filter.config.options.value;
+            if (typeof filter.config.options.value === 'string') {
+              value = ctx.interpolate(value);
+            }
```
Comment (samjewell) [inline packages/grafana-data/src/transformations/transformers/filterByValue.ts:85]: Could it be worth adding an extra test to protect this functionality? As we're adding an extra code-branch here 🙏

**Comment by nmarrs** [comment]
> Looks great, thanks for this change. Sorry to ask - does this support Multi-select / select-All?

Hey @samjewell - what do you mean by whether or not this supports multi-select / select-all? In terms of the regex or selecting many series at a time?

File: packages/grafana-data/src/transformations/transformers/filterByValue.ts (hunk 1 above)
Comment (nmarrs) [reply packages/grafana-data/src/transformations/transformers/filterByValue.ts]: done :)

**Review by adela-almasan — APPROVED** [review]

**Comment by samjewell** [comment]
> **Error building instance**: Contact #proj-ephemeral-hg-instances if it is not a compile error. [Logs](https://github.com/grafana/grafana/actions/runs/10169792941)

@nmarrs is this a problem with your code? Or something else going wrong?

**Comment by nmarrs** [comment]
> > **Error building instance**: Contact #proj-ephemeral-hg-instances if it is not a compile error. [Logs](https://github.com/grafana/grafana/actions/runs/10169792941)
> 
> @nmarrs is this a problem with your code? Or something else going wrong?

Hmm looking at the build log it seems unrelated to these code changes

`Error: wire: /src/pkg/extensions/pluginsintegration/pluginsintegration.go:47:2: first argument to Bind must be a pointer to an interface type; found <nil>`
Screenshot 2024-07-30 at 2 35 07 PM

File: public/app/features/transformers/FilterByValueTransformer/ValueMatchers/RegexMatcherEditor.tsx [hunk 2]
```diff
@@ -37,6 +57,18 @@ export function regexMatcherEditor(
       [options, onChange, isInvalid, field, converter]
     );
 
+    if (cfg.featureToggles.transformationsVariableSupport) {
+      return (
+        <SuggestionsInput
```
Comment (nmarrs) [inline public/app/features/transformers/FilterByValueTransformer/ValueMatchers/RegexMatcherEditor.tsx:62]: shout out to @adela-almasan for implementing this UX improvement 🥇

**Pull request merged**